src/pages/SignIn.js

```diff
@@ -1,30 +1,35 @@
 import { useState, useRef, useEffect } from "react";
+import useRedirectLink from "../hooks/useRedirectLink";
 
 import strings from "../i18n/definitions";
 
 import * as s from "../components/FormPage.sc";
 import LocalStorage from "../assorted/LocalStorage";
 
-export default function SignIn({ api, signInAndRedirect, setRedirectLink }) {
+export default function SignIn({ api, handleSuccessfulSignIn }) {
   // TODO: Fix this bug in a different way. Requires understanding why strings._language changes to "da" without it being asked to, whenever this component renders. Perhaps it imports an un-updated version of strings?
   strings.setLanguage(LocalStorage.getUiLanguage().code);
 
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [errorMessage, setErrorMessage] = useState("");
 
+  let { handleRedirect } = useRedirectLink();
+
   let emailInputDOM = useRef();
+
   useEffect(() => {
     emailInputDOM.current.focus();
-    const queryParameters = new URLSearchParams(window.location.search);
-    setRedirectLink(queryParameters.get("redirectLink"));
   }, []);
 
   function handleSignIn(e) {
     e.preventDefault();
     api.signIn(email, password, setErrorMessage, (sessionId) => {
       api.getUserDetails((userInfo) => {
-        signInAndRedirect(userInfo);
+        handleSuccessfulSignIn(userInfo);
+        /* If a redirect link exists, uses it to redirect the user, 
+        otherwise, uses the location from the function argument. */
+        handleRedirect("/articles");
```

Comment:
It's great that we have a comment, but it would be nice if the function name would itself be more self-understandable. Something like `handleRedirectLinkOrGoTo("/articles")`? Or maybe `unlessRedirectedGoTo("/articles")` ?

Reply:
✅ This function is now renamed to `handleRedirectLinkOrGoTo("/articles")`